controllers/imagejob/imagejob_controller.go

```diff
@@ -151,12 +168,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			ObjectMeta: metav1.ObjectMeta{Namespace: "eraser-system", Name: podName, Labels: map[string]string{"name": image.Name}},
 		}
 
-		// TODO: check if pod fits and can be scheduled on node
-		err = r.Create(ctx, pod)
-		if err != nil {
-			return ctrl.Result{}, err
+		// Check if pod fits and can be scheduled on node
+		fitness := checkNodeFitness(pod, &n)
+
+		if fitness {
+			err = r.Create(ctx, pod)
+			if err != nil {
+				return ctrl.Result{}, err
```

Comment:
Should this stop all other reconciliation?

Reply:
Yes